src/lib/transaction_pool/f_sequence.ml

```diff
@@ -0,0 +1,699 @@
+open Core
+
+(** A digit is a container of 1-4 elements. *)
+module Digit = struct
+  (* We use GADTs to track whether it's valid to remove/add an element to a
+     digit, which gets us type safe (un)cons and (un)snoc. *)
+
+  type addable = Type_addable
+
+  type not_addable = Type_not_addable
+
+  type removable = Type_removable
+
+  type not_removable = Type_not_removable
+
+  (* Silence the unused value warnings. Y U NO EmptyDataDecls, Ocaml? *)
```

Comment:
I suppose the OCamlish way to suppress these warnings is to bracket the type definitions with
```
  [@@@warning "-37"]
  ...
  [@@@warning "+37"]
```

Reply:
meaning, one bracket for all the definitions, not one bracket per definition